RandomForest_v9.ipynb

```diff
@@ -0,0 +1,354 @@
+    "clean_df = pd.DataFrame(imp.fit_transform(df.loc[:, \"ADJP_->_JJ_x\":\"zcr_var_y\"]), columns = list(df)[df.columns.get_loc(\"ADJP_->_JJ_x\"):(df.columns.get_loc(\"zcr_var_y\") + 1)])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#append new dataframe with selected columns\n",
+    "clean_df = clean_df.join([df['subject_id'], df['firstname'], df['age_at_record'], df['year_diag'], df['group_label_id']])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#get list of subject names\n",
+    "subj_names = df.iloc[:, -1].dropna().unique().tolist()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#select feature names\n",
+    "man_feat = list(df)[df.columns.get_loc(\"ADJP_->_JJ_x\"):(df.columns.get_loc(\"zcr_var_x\") + 1)]\n",
+    "asr_feat = list(df)[df.columns.get_loc(\"ADJP_->_JJ_y\"):(df.columns.get_loc(\"zcr_var_y\") + 1)]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#shuffle rows for better classification\n",
+    "clean_df = shuffle(clean_df)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#pick random forest for classifier\n",
+    "clf = RandomForestClassifier(n_estimators = 2, max_depth = 70)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#UPDATED - for SAMPLE - LOOCV function\n",
+    "def sample_transcript_loocv(subject_names, classifier_name, clf, df_clean, featureset):\n",
+    "    prediction = []\n",
+    "    y_test = []\n",
+    "    for name in subject_names:        \n",
+    "        df_data = df_clean.copy()\n",
+    "        df_data = df_data[df_data.firstname.isin(subject_names)]\n",
+    "        ind_train = [~df_data['firstname'].isin([name])]\n",
+    "        ind_test = [df_data['firstname'].isin([name])]\n",
+    "        df_data = df_data.sort_values('age_at_record')\n",
+    "        #for every train-test split, obtain means only from training HC group (so that model is completely agnostic to labels)\n",
+    "        X = df_data[featureset].values\n",
+    "        y_label = df_data['group_label_id'].values\n",
+    "        years = df_data['age_at_record'].values\n",
+    "        years_diag = df_data['year_diag'].values\n",
+    "        sub_id = df_data['subject_id'].values\n",
+    "        X = StandardScaler().fit_transform(X)\n",
+    "        sel = VarianceThreshold(threshold=(.99 * (1 - .99)))\n",
+    "        X = sel.fit_transform(X)\n",
+    "        ind_train = [~df_data['firstname'].isin([name])]\n",
+    "        ind_test = [df_data['firstname'].isin([name])] \n",
```

Comment:
Lines 172 and 173 are the same as 160-161. Delete either the first ones or the second.